src/Symfony/Component/Messenger/Transport/KernelTerminateTransportFactory.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Messenger\Transport;
+
+use Psr\Container\ContainerInterface;
+use Symfony\Component\Messenger\Exception\InvalidArgumentException;
+
+class KernelTerminateTransportFactory implements TransportFactoryInterface
+{
+    private $busLocator;
+
+    public function __construct(ContainerInterface $busLocator)
+    {
+        $this->busLocator = $busLocator;
+    }
+
+    public function createTransport(string $dsn, array $options): TransportInterface
+    {
+        if (false === $parsedUrl = parse_url($dsn)) {
+            throw new InvalidArgumentException(sprintf('The given kernel.terminate DSN "%s" is invalid.', $dsn));
+        }
+
+        $parsedQuery = array();
+        parse_str($parsedUrl['query'] ?? null, $parsedQuery);
+        $busId = $parsedQuery['bus'] ?? $options['bus'] ?? null;
+
+        if (null === $busId) {
+            throw new InvalidArgumentException(sprintf('Missing mandatory "bus" option for kernel.terminate transport with DSN "%s"', $dsn));
+        }
+
+        if (!$this->busLocator->has($busId)) {
+            throw new InvalidArgumentException(sprintf('No bus was found with id "%s" for kernel.terminate transport with DSN "%s"', $busId, $dsn));
+        }
+
+        return new MemoryTransport($this->busLocator->get($busId));
+    }
+
+    public function supports(string $dsn, array $options): bool
+    {
+        return 'symfony://kernel.terminate' === $dsn;
```

Comment:
Should we also change this to strpos?

Reply:
Indeed, forgot to update that. Good catch. I let you take back control of your PR :)

Reply:
Ha ha, ok.
I’ll wait for sroze feedback before creating all the tests.